### PR TITLE
Replace hardcoded urls in contest card 

### PR DESCRIPTION
**Pull request by msmner**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/257

### CODE REVIEW

**Review by mishogv — APPROVED** [review]

**Review by PetarNeshkov — APPROVED** [review]

**Review by mishogv — APPROVED** [review]

**Review by shsimeonova — APPROVED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/contest-card/ContestCard.tsx [hunk 1]
```diff
@@ -34,6 +37,8 @@ const ContestCard = ({ contest }: IContestCardProps) => {
     const contestCardControlBtns = 'card-control-buttons';
     const contestCardControlBtnsClassName = concatClassNames(styles.contestCardControls, contestCardControlBtns);
 
+    const { getRegisterContestTypeUrl } = useUrls();
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/contest-card/ContestCard.tsx:40]: This should be in a new hook called `useAppUrls` or something.

`useUrls` is for API calls only.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/contest-card/ContestCard.tsx [hunk 2]
```diff
@@ -82,7 +94,7 @@ const ContestCard = ({ contest }: IContestCardProps) => {
             <div className={contestCardControlBtnsClassName}>
                 <LinkButton
                   id="button-card-compete"
-                  to={`/contests/${id}/register/compete`}
+                  to={getRegisterContestTypeUrl({ id, participationType } as IRegisterForContestTypeUrlParams)}
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/contest-card/ContestCard.tsx:97]: This cast should be unnecessary

**Review by Minkov — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/contest-card/ContestCard.tsx (hunk 2 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/contest-card/ContestCard.tsx:97]: Fixed in another branch.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/contest-card/ContestCard.tsx (hunk 1 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/contest-card/ContestCard.tsx:40]: Resolved in a different issue.

**Pull request merged**